narwhals/_arrow/utils.py

```diff
@@ -494,3 +496,43 @@ def arange(start: int, end: int, step: int) -> ArrayAny:
         return pa.array(np.arange(start, end, step))
     # NOTE: Added in https://github.com/apache/arrow/pull/46778
     return pa.arange(start, end, step)  # type: ignore[attr-defined]
+
+
+def list_agg(
+    array: ChunkedArrayAny,
+    func: Literal["min", "max", "mean", "approximate_median", "sum"],
+) -> ChunkedArrayAny:
+    version = sys.version_info
+    if func == "approximate_median" and version < (3, 10):  # pragma: no cover
+        msg = (
+            f"The minimum supported Python version for {func} is 3.10."
+            f"\nGot: {version.major}.{version.minor}.{version.micro}."
+        )
+        raise NotImplementedError(msg)
```

Comment:
I double checked what's happening here - I was a bit skeptical because there is no error for `{Series, Expr}.median()`.

For pyarrow itself this is not an issue, you can remove the exception and the tests will pass. Then also `pandas` (and `pandas[pyarrow]`) will pass (assuming pandas>=2.2).

So in summary, I am just a bit confused about this error and when it shows up. I think we should try to remove it and see what happens on CI

Reply:
Thanks @FBruzzesi ! Yes it seems that there is no need for the minimum Python version and the tests pass. The current CI failures seem to be unrelated to this PR